### PR TITLE
Run post deploy tests from teamcity

### PR DESCRIPTION
**Pull request by johnduffell**
## What are you doing in this PR?

This PR adds a script so we can run the post deploy tests from teamcity.
We also need to:
- [x] fix broken tests
- [x] actually add the build to TC - including any env vars we need to resynthesise
- [x] clean up any old travis stuff as needed
- [x] update wiki description of post deploy tests

https://trello.com/c/Btdw9LKy/3831-get-post-deploy-tests-running

## Why are you doing this?

The tests are broken, and we don't want to fix them in travis as that doesn't seem to be the popular way nowadays.

The riffraff hooks may work to call a url, and you can kick off a build using curl for teamcity, but there were difficulties supplying the headers for auth (and making sure they stayed secure enough) so we eventually settled on running on a schedule from teamcity for now.  This can be improved in future if we like.

### CODE REVIEW

File: prout/seen.md [hunk 1]
```diff
@@ -1,2 +1,2 @@
 
-- Keep an eye out for [post deployment test](https://github.com/guardian/support-frontend/wiki/Post-Deployment-Testing) results on [Travis](https://travis-ci.org/guardian/support-frontend/builds)
+- Keep an eye out for [post deployment test](https://github.com/guardian/support-frontend/wiki/Post-Deployment-Testing) results on [TOBE CONFIRMED URL](https://support.thegurardian.com)
```
Comment (johnduffell) [inline prout/seen.md:2]:
```suggestion
- Keep an eye out for [post deployment test](https://github.com/guardian/support-frontend/wiki/Post-Deployment-Testing) results on [teamcity](https://teamcity.gutools.co.uk/buildConfiguration/memsub_Support_SupportFrontendPostDeployTests?branch=%3Cdefault%3E&mode=builds)
```

File: support-frontend/scripts/post_test_results_to_chat.sh [hunk 2]
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
-
-curl -X POST -H 'Content-Type: application/json' "${GOOGLE_CHAT_WEB_HOOK}" -d '{"text": "❌ The post deployment tests for support frontend have failed! <users/all> \n \n 👉 <https://travis-ci.org/guardian/support-frontend|Travis build> \n 🤖 <https://automate.browserstack.com/dashboard/v2/builds/31f35a1d9bccc9d45360aa7bfd651fcd9e1499d0|Browser stack test results> \n \n 📖 <https://github.com/guardian/support-frontend/wiki/Post-deployment-test-runbook|Check the runbook for a step by step guide>"}'
+echo "Tests have failed - calling webhook"
+curl -X POST -H 'Content-Type: application/json' "${GOOGLE_CHAT_WEB_HOOK}" -d '{"text": "THIS THIS A TEST PLEASE IGNORE! \n❌ The post deployment tests for support frontend have failed! <users/all> \n \n 👉 <https://travis-ci.org/guardian/support-frontend|Travis build> \n 🤖 <https://automate.browserstack.com/dashboard/v2/builds/31f35a1d9bccc9d45360aa7bfd651fcd9e1499d0|Browser stack test results> \n \n 📖 <https://github.com/guardian/support-frontend/wiki/Post-deployment-test-runbook|Check the runbook for a step by step guide>"}'
```
Comment (johnduffell) [inline support-frontend/scripts/post_test_results_to_chat.sh:3]:
```suggestion
curl -X POST -H 'Content-Type: application/json' "${GOOGLE_CHAT_WEB_HOOK}" -d '{"text": "❌ The post deployment tests for support frontend have failed! <users/all> \n \n 👉 <https://teamcity.gutools.co.uk/buildConfiguration/memsub_Support_SupportFrontendPostDeployTests?branch=%3Cdefault%3E&mode=builds|Travis build> \n 🤖 <https://automate.browserstack.com/dashboard/v2/builds/31f35a1d9bccc9d45360aa7bfd651fcd9e1499d0|Browser stack test results> \n \n 📖 <https://github.com/guardian/support-frontend/wiki/Post-deployment-test-runbook|Check the runbook for a step by step guide>"}'
```

File: support-frontend/test/selenium/subscriptions/CheckoutsSpec.scala [hunk 3]
```diff
@@ -36,6 +36,7 @@ class CheckoutsSpec extends AnyFeatureSpec
   Feature("Digital Pack checkout") {
     Scenario("User already logged in - Stripe checkout") {
       testCheckout("Digital Pack", new DigitalPackCheckout, new DigitalPackProductPage, payWithStripe)
+      fail()
```
Comment (johnduffell) [inline support-frontend/test/selenium/subscriptions/CheckoutsSpec.scala:39]:
```suggestion
```

File: support-frontend/scripts/post_test_results_to_chat.sh [hunk 4]
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
-
-curl -X POST -H 'Content-Type: application/json' "${GOOGLE_CHAT_WEB_HOOK}" -d '{"text": "❌ The post deployment tests for support frontend have failed! <users/all> \n \n 👉 <https://travis-ci.org/guardian/support-frontend|Travis build> \n 🤖 <https://automate.browserstack.com/dashboard/v2/builds/31f35a1d9bccc9d45360aa7bfd651fcd9e1499d0|Browser stack test results> \n \n 📖 <https://github.com/guardian/support-frontend/wiki/Post-deployment-test-runbook|Check the runbook for a step by step guide>"}'
+echo "Tests have failed - calling webhook"
+curl -X POST -H 'Content-Type: application/json' "${GOOGLE_CHAT_WEB_HOOK}" -d '{"text": "❌ The post deployment tests for support frontend have failed! <users/all> \n \n 👉 <https://teamcity.gutools.co.uk/buildConfiguration/memsub_Support_SupportFrontendPostDeployTests?branch=%3Cdefault%3E&mode=builds|Travis build> \n 🤖 <https://automate.browserstack.com/dashboard/v2/builds/31f35a1d9bccc9d45360aa7bfd651fcd9e1499d0|Browser stack test results> \n \n 📖 <https://github.com/guardian/support-frontend/wiki/Post-deployment-test-runbook|Check the runbook for a step by step guide>"}'
```
Comment (johnduffell) [inline support-frontend/scripts/post_test_results_to_chat.sh:3]:
```suggestion
curl -X POST -H 'Content-Type: application/json' "${GOOGLE_CHAT_WEB_HOOK}" -d '{"text": "❌ The post deployment tests for support frontend have failed! <users/all> \n \n 👉 <https://teamcity.gutools.co.uk/buildConfiguration/memsub_Support_SupportFrontendPostDeployTests?branch=%3Cdefault%3E&mode=builds|Teamcity build> \n 🤖 <https://automate.browserstack.com/dashboard/v2/builds/31f35a1d9bccc9d45360aa7bfd651fcd9e1499d0|Browser stack test results> \n \n 📖 <https://github.com/guardian/support-frontend/wiki/Post-deployment-test-runbook|Check the runbook for a step by step guide>"}'
```

File: support-frontend/test/selenium/util/Config.scala [hunk 5]
```diff
@@ -33,15 +30,12 @@ object Config {
...
-  val paypalBuyerEmail = conf.getString("paypal.sandbox.buyer.email")
-
-  val paypalBuyerPassword = conf.getString("paypal.sandbox.buyer.password")
-
   def printSummary(sessionId: SessionId): Unit = {
     logger.info("Selenium Test Configuration")
     logger.info("=============================")
     logger.info(s"Stage: ${conf.getString("stage")}")
     logger.info(s"Support Frontend: ${supportFrontendUrl}")
+    logger.info(s"Client secret: *****${idapiClientAccessTokenSecret.substring(5)}")
```
Comment (johnduffell) [inline support-frontend/test/selenium/util/Config.scala:38]: do we still need this?

Comment (rupertbates) [reply support-frontend/test/selenium/util/Config.scala:38]: No that could go

**Review by rupertbates — APPROVED** [review]

File: support-frontend/test/selenium/util/Config.scala (hunk 5 above)
Comment (johnduffell) [reply support-frontend/test/selenium/util/Config.scala:38]:
```suggestion
```

**Comment by i-hardy** [comment]
👏 👏 👏

**Pull request merged**